src/state/index.ts

```diff
@@ -0,0 +1,45 @@
+import { Key } from "../models/key.model";
+
+type Controller<T> = (state: T[], current: T) => void;
```

Comment:
There can be just one controller for this simple UI. might not require multiple controllers actually.